src/mds/MDSRank.cc

```diff
@@ -2137,7 +2137,7 @@ void MDSRank::active_start()
 
   dout(10) << __func__ << ": initializing metrics handler" << dendl;
   metrics_handler.init();
-  messenger->add_dispatcher_tail(&metrics_handler);
+  messenger->add_dispatcher_head(&metrics_handler, 20); // very high but behind Beacon
```

Comment:
Still magic numbers. Now, one needs to go find where beacon handler is added and check that number. And if that number changes, how do they know to update this place?

If this should be high but behind beacon, this should be 

```
  messenger->add_dispatcher_tail(&metrics_handler, ::priority_high);
```

and beacon should stay 

```
  messenger->add_dispatcher_head(&beacon, ::priority_high);
```

Reply:
fixed